### PR TITLE
fix bug causing error when Marker#setLngLat is called on a marker that's already on the map

### PR DESCRIPTION
**Pull request by mollymerp**
## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] document any changes to public APIs
 - [ ] post benchmark scores
 - [x] manually test the debug page

also adds a test that would have caught the bug.

I decided to go back to the approach where we don't add in an extra function argument to the `_update` function. IMO this reduces unnecessary complexity bc the `Evented` object that gets passed gives us all the information we need to produce the desired functionality. This code is less explicit than with `roundedPos`, but I added a detailed comment to compensate for this. 

Thanks @runetvilum for catching this bug!  ref #3283

cc @jfirebaugh @lucaswoj

### CODE REVIEW

File: js/ui/marker.js [hunk 1]
```diff
@@ -66,6 +66,7 @@ Marker.prototype = {
         if (this._map) {
             this._map.off('click', this._onMapClick);
             this._map.off('move', this._update);
+            this._map.on('moveend', this._update);
```
Comment (jfirebaugh) [inline js/ui/marker.js]: `on` ⇢ `off`

File: test/js/ui/marker.test.js [hunk 2]
```diff
@@ -29,6 +29,14 @@ test('Marker', function (t) {
         t.end();
     });
 
+    t.test('marker\'s lngLat can be changed', function (t) {
+        var map = createMap();
+        var marker = new Marker(window.document.createElement('div')).setLngLat([-77.01866, 38.888]).addTo(map);
+        t.ok(marker.setLngLat([-76,39]) instanceof Marker, 'marker.setLngLat() returns Marker instance');
+        t.ok(marker._lngLat.lng === -76 &&  marker._lngLat.lat === 39, 'marker\'s position can be updated');
```
Comment (jfirebaugh) [inline test/js/ui/marker.test.js]: Use `marker.getLngLat()` rather than a private property.

**Comment by ajfickas** [comment]
@mollymerp When will this fix be officially released?

I've got an app we've been migrating to Mapbox GL JS which we're planning to release hopefully this week and this fix looks like it will address a few show-stopper bugs we're seeing. Specifically, `marker.setLngLat` throwing an error when called during a map move event (to allow dragging a marker on the map), and markers "jiggling" when panning the map.

(Didn't see a release schedule out there, so this was the best place I found to ask. Let me know if there a better place to get an answer!)

**Comment by mollymerp** [comment]
@ajfickas sorry for the inconvenience! 

We track our releases at www.github.com/mapbox/mapbox-gl-js/milestones and our next one is scheduled for this Wednesday. There is a chance we won't get it out on Wednesday if there are outstanding release blockers, but we are also eager to get another release out ASAP.

**Comment by ajfickas** [comment]
Thanks for the quick response, that answers my question! I'll stay tuned!

**Pull request merged**